ios/browser/api/certificate/brave_certificate.mm

```diff
@@ -0,0 +1,187 @@
+      return nullptr;
+    }
+
+    net::CertErrors errors;
+    extended_cert_ =
+        scoped_refptr<net::ParsedCertificate>(net::ParsedCertificate::Create(
+            std::move(cert_buffer),
+            net::x509_util::DefaultParseCertificateOptions() /* {} */,
+            &errors));
+
+    if (!extended_cert_) {
+      NSLog(@"Error: %s\n", errors.ToDebugString().c_str());
+      return nullptr;
+    }
+
+    [self parseCertificate];
+
+    cert_data_.reset();
+    extended_cert_.reset();
+    public_key_.reset();
+  }
+  return self;
+}
+
+- (void)dealloc {
+  extended_cert_.reset();
+}
+
+- (void)parseCertificate {
+  _isRootCertificate = [self is_root_certificate];
+  _isCertificateAuthority = [self is_certificate_authority];
+  _isSelfSigned = [self is_self_signed];
+  _isSelfIssued = [self is_self_issued];
+  _subjectName = [[BraveCertificateRDNSequence alloc]
+      initWithBERName:extended_cert_->tbs().subject_tlv
+             uniqueId:extended_cert_->tbs().subject_unique_id];
+  _issuerName = [[BraveCertificateRDNSequence alloc]
+      initWithBERName:extended_cert_->tbs().issuer_tlv
+             uniqueId:extended_cert_->tbs().issuer_unique_id];
+  _serialNumber = base::SysUTF8ToNSString(brave::hex_string_from_string(
+      extended_cert_->tbs()
+          .serial_number.AsString()));  // base::HexEncode(data, length)
+  _version = static_cast<std::int32_t>(extended_cert_->tbs().version) +
+             1;  // version + 1
+  _signature = [[BraveCertificateSignature alloc]
+      initWithCertificate:extended_cert_.get()];
+  _notValidBefore = brave::date_to_ns(x509_utils::generalized_time_to_time(
+      extended_cert_->tbs().validity_not_before));
+  _notValidAfter = brave::date_to_ns(x509_utils::generalized_time_to_time(
+      extended_cert_->tbs().validity_not_after));
+  _publicKeyInfo = [[BraveCertificatePublicKeyInfo alloc]
+      initWithCertificate:extended_cert_.get()
+                  withKey:public_key_.get()];
+  //  _extensions = nullptr;
+  _sha1Fingerprint = [[BraveCertificateFingerprint alloc]
+      initWithCertificate:cert_data_.get()
+                 withType:BraveFingerprintType_SHA1];
+  _sha256Fingerprint = [[BraveCertificateFingerprint alloc]
+      initWithCertificate:cert_data_.get()
+                 withType:BraveFingerprintType_SHA256];
+
+  [self parseExtensions];
+}
+
+- (bool)is_root_certificate {
+  net::CertErrors errors;
+  std::string normalized_subject;
+  net::der::Input subject_value = extended_cert_->normalized_subject();
+  if (!net::NormalizeName(subject_value, &normalized_subject, &errors)) {
+    return false;
+  }
+
+  std::string normalized_issuer;
+  net::der::Input issuer_value = extended_cert_->normalized_issuer();
+  if (!net::NormalizeName(issuer_value, &normalized_issuer, &errors)) {
+    return false;
+  }
+
+  return normalized_subject == normalized_issuer;
+}
+
+- (bool)is_certificate_authority {
+  // return X509_check_ca(x509_cert_);
+  return false;
```

Comment:
This on purpose?

Reply:
Yes, for now as I haven't parsed extensions to check if the cert is a CA yet.